### PR TITLE
[Fixes #4223] Added a facade for ViewComponentResult

### LINKED ISSUE #4223: ServiceLocator: ViewViewComponentResult needs facade with executer

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewComponentResultExecutor.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    {
+        private readonly HtmlEncoder _htmlEncoder;
+        private readonly HtmlHelperOptions _htmlHelperOptions;
+        private readonly ILogger<ViewComponentResult> _logger;
+        private readonly IModelMetadataProvider _modelMetadataProvider;
+        private readonly ITempDataDictionaryFactory _tempDataDictionaryFactory;
+        private readonly IViewComponentHelper _viewComponentHelper;
+
+        public ViewComponentResultExecutor(
+            IOptions<MvcViewOptions> mvcHelperOptions,
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewComponentResultExecutor.cs]: `IOptions<MvcViewOptions> mvcViewOptions`

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewComponentResultExecutor.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                viewData = new ViewDataDictionary(_modelMetadataProvider, context.ModelState);
+            }
+
+            var tempData = viewComponentResult.TempData;
+            if (tempData == null)
+            {
+                tempData = _tempDataDictionaryFactory.GetTempData(context.HttpContext);
+            }
+
+            string resolvedContentType = null;
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewComponentResultExecutor.cs]: Shouldn't need to initialize these two variables.

**Comment by pranavkm** [comment]
:shipit:

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponentResult.cs [hunk 3]
```diff
@@ -68,83 +59,11 @@ public class ViewComponentResult : ActionResult
         public IViewEngine ViewEngine { get; set; }
 
         /// <inheritdoc />
-        public override async Task ExecuteResultAsync(ActionContext context)
-        {
-            var response = context.HttpContext.Response;
+        public override Task ExecuteResultAsync(ActionContext context)
+        {            
             var services = context.HttpContext.RequestServices;
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponentResult.cs]: Suggest `null` checks for `context` (`ArgumentNullException`), `context.HttpContext` (`ArgumentException(Resources.FormatPropertyOfTypeCannotBeNull(...))`), and `services`.

**Pull request merged**